lib/endpoints/volumes.js

```diff
@@ -76,16 +98,37 @@ function getVolume(req, res, next) {
     });
 }
 
+//
+// NOTE: This will add the 'listVolumesPredicate' property to the 'req' object
+// if req.query.params.predicate can be turned into a valid predicate.
+//
 function validateListVolumesInput(req, res, next) {
     assert.object(req, 'req');
+    assert.object(req.params, 'req.params');
+    assert.object(req.query, 'req.query');
     assert.object(res, 'res');
     assert.func(next, 'next');
 
-    var validationErrs = [];
+    var invalidParamsErrs;
     var latestErr;
+    var predFields;
+    var validationErrs = [];
+    var VALID_PARAM_NAMES = [
+        'name',
+        'predicate',
+        'size',
+        'state',
+        'type'
+    ];
+
+    // Use req.query since that doesn't include 'account' which comes from the
+    // URL rather than query parameters.
+    invalidParamsErrs = checkInvalidParams(req.query, VALID_PARAM_NAMES);
+    validationErrs = validationErrs.concat(invalidParamsErrs);
 
     if (req.params.name) {
```

Comment:
I know this was already a problem here with the previous version of this file, but we should use `if (req.params.name !== undefied) {` instead of implicitly converting `req.params.name` to a boolean expression. Otherwise, an empty string or anything that implicitly coerces to a boolean value of `false` will not be validated.

We should use the same approach for all similar tests in this function (e.g with `req.params.size` below, etc.).